test/ibmq/test_experiment.py

```diff
@@ -526,6 +526,33 @@ def test_upload_update_plot_data(self):
         rplot = self.provider.experiment.retrieve_plot(new_exp.uuid, plot_name)
         self.assertEqual(rplot, friend_bytes, "Retrieved plot not equal updated plot.")
 
+    def test_experiments_limit(self):
+        """Test getting experiments with a limit."""
+        limits = [10, 1000, 1001]
+
+        for limit in limits:
+            with self.subTest(limit=limit):
+                experiments = self.provider.experiment.experiments(limit=limit)
+                self.assertEqual(len(experiments), limit)
+                self.assertEqual(len({expr.uuid for expr in experiments}), len(experiments))
+
+        with self.assertRaises(ValueError) as context_manager:
+            self.provider.experiment.experiments(limit=-1)
+        self.assertIn("limit", str(context_manager.exception))
+
+    def test_analysis_results_limit(self):
+        """Test getting analysis results with a limit."""
+        limits = [10, 1000, 1001]
```

Comment:
1001 is going to be capped at 1000 on the server side, per the docs:

https://github.ibm.com/IBM-Q-Software/sw-ibmq-results/blob/staging/docs/api-ref.md#pagination

>limit: Restrict the result set to this many records. Must be at least 1 and at most the configured maximum for the server (default: 1000). If the limit is reached a marker field is returned in the response body which can be be provided using the marker query parameter in subsequent requests.

So how is the assertion on L549 not failing?

Reply:
Oh I get it, it's using the marker to page on the client side, so it's really making two requests, the first is limited to 1000 results and then it uses the marker to get the next page which is going to limit to 1 result from the last marker. Nice.